### PR TITLE
CI - Add changelog and deploy-changelog action

### PR DESCRIPTION
**Pull request by da1910**
#227 

Add the changelog config as per the migration guide for actions v6

### CODE REVIEW

**Comment by Andy-Grigg** [comment]
@da1910 Does this address https://github.com/ansys/grantami-recordlists/issues/227?

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -158,10 +158,22 @@ jobs:
...
+    name: "Update CHANGELOG for new tag"
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: ansys/actions/doc-deploy-changelog@v6
+        with:
+          token: ${{ '{{ secrets.PYANSYS_CI_BOT_TOKEN }}' }}
```
Comment (Andy-Grigg) [inline .github/workflows/ci_cd.yml:171]: This seems strangely indirect. Is this required?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+This project uses [towncrier](https://towncrier.readthedocs.io/) and the
+changes for the upcoming release can be found in
+<https://github.com/ansys/openapi-common/tree/main/doc/changelog.d/>.
```
Comment (Andy-Grigg) [inline CHANGELOG.md:5]: Wrong repo?

**Review by Andy-Grigg — CHANGES_REQUESTED** [review]
One required change, one question.

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (da1910) [reply .github/workflows/ci_cd.yml:171]: https://actions.docs.ansys.com/version/stable/migrations/docs-deploy-changelog-setup.html#docs-deploy-changelog-action-setup copied verbatim from the migration guide

**Review by Andy-Grigg — APPROVED** [review]

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (da1910) [reply .github/workflows/ci_cd.yml]: Checked with Kerry and it's an oversight

**Pull request merged**